src/pages/SupplyRequisitionPage.js

```
@@ -0,0 +1,362 @@
+              if (newComment !== this.props.requisition.comment) {
+                this.props.database.write(() => {
+                  this.props.requisition.comment = newComment;
+                  this.props.database.save('Requisition', this.props.requisition);
+                });
+              }
+              this.closeModal();
+            }}
+          />
+          );
+    }
+  }
+
+  renderButtons() {
+    return (
+      <View style={globalStyles.pageTopRightSectionContainer}>
+        <View style={globalStyles.verticalContainer}>
+          <PageButton
+            style={globalStyles.topButton}
+            text={buttonStrings.use_requested_quantity}
+            onPress={this.onUseRequestedQuantity}
+            isDisabled={this.props.requisition.isFinalised}
+          />
+        </View>
+      </View>
+    );
+  }
+
+  render() {
+    return (
+      <GenericPage
+        data={this.state.data}
+        refreshData={this.refreshData}
+        renderCell={this.renderCell}
+        renderTopLeftComponent={this.renderPageInfo}
+        renderTopRightComponent={this.renderButtons}
+        onEndEditing={this.onEndEditing}
+        onSelectionChange={this.onSelectionChange}
+        defaultSortKey={this.dataFilters.sortBy}
+        defaultSortDirection={this.dataFilters.isAscending ? 'ascending' : 'descending'}
+        columns={[
+          {
+            key: 'itemCode',
+            width: 1.5,
+            title: tableStrings.code,
+            sortable: true,
+          },
+          {
+            key: 'itemName',
+            width: 4,
+            title: tableStrings.item_name,
+            sortable: true,
+          },
+          {
+            key: 'ourStockOnHand', // TODO: should this column name be consistent with other pages
+            width: 1.5,
+            title: tableStrings.our_stock,
+            sortable: true,
+            alignText: 'right',
+          },
+          {
+            key: 'stockOnHand',
```

I'm wondering if it is a good practice to throw errors rather than return null with such checks else where. Would make bugsnag logs be more informative!